### ISSUE TITLE
Add all preset keymaps

### ISSUE THREAD

**Issue by mondalaci** [UltimateHackingKeyboard/agent#91]
The following keymaps are planned to be added:
- QWERTY
- Dvorak
- Colemak
- Workman
- Programmer's Dvorak
- Bepo
- Neo
- Neo2
- Egghead.io
- VIM
- Emacs
- [Oyayubi shift](https://en.wikipedia.org/wiki/Thumb-shift_keyboard)

For every keymap there will be a PC and Mac version, such as:
- QWERTY for PC
- QWERTY for Mac

See https://en.wikipedia.org/wiki/Keyboard_layout regarding these layouts and http://xahlee.info/kbd/dvorak_and_all_keyboard_layouts.html regarding some even more exotic layouts.

I don't insist to these exact layouts but I think they would make sense to be featured.

Switch keymap actions will be bound to the Fn layer so that the QWERTY for PC keymap will be activated by Fn+Q, the QWERTY for Mac keymap will be activated by Fn+W, and such.

Once the key action editor will fully work we should be able to crowdsource this issue. I plan mention it in an upcoming newsletter then.

**Comment by NejcZdovc** [comment]
This issue can be done when #93 is finished.

**Comment by NejcZdovc** [comment]
For release this are relevant:

- QWERTY
- Dvorak
- Colemak
- Workman

Everything else is nice to have.

**Comment by NejcZdovc** [comment]
Blocked by #199

**Comment by fjozsef** [comment]
Blocked #136, #286 and #287

**Comment by fjozsef** [comment]
@mondalaci It should be ready to go! :)

**Comment by Jos512** [comment]
Cool that you guys are going to support those different keymaps! :)

**Comment by TorC8** [comment]
It'll probably be a bit before some of those can be supported.  Off the top of my head, I believe QWERTY, Dvorak, Colemak, and Workman are easy, and are likely to be rounded out once the pilot run gets in people's hands so they can actively work on the layouts.  I'm less sure of the details of Emacs and VIM layouts, but these may well fit in this category too.

Programmer's Dvorak and Bépo don't just move whole keys around, but change symbol pairings.  This require more trickery in the keyboard to manage, namely being able to send a different scancode based on whether the shift key is pressed.  ETA on this feature is unknown/not yet in the pipeline, last I heard.

The remainder of those listed I am unfamiliar with.

**Comment by mondalaci** [comment]
@Jos512 Glad you like them! Although we may only support the most popular ones from the get go. Luckily, you'll be able to create your own keymaps. :)

@TorC8 Good points! We'll probably only implement the ones you suggested for the pilot run. You're also right that some of the features for the more exotic layouts are not in place yet.

**Comment by mondalaci** [comment]
Done by e99fb08cc2d5aa1b640322dbc9b84bc34254a4bd.

**Comment by Jos512** [comment]
Nice to see the Agent coming along. Looking forward to use it!
